consonance-client/src/main/java/io/consonance/client/cli/Main.java

```diff
@@ -0,0 +1,339 @@
+                for(String extraFile : extraFiles){
+                    String[] values = extraFile.split("=");
+                    final int lengthOfValues = 3;
+                    if (values.length != lengthOfValues){
+                        kill("consonance: failure parsing: '%s'.", extraFile);
+                    }
+                    ExtraFile file =  new ExtraFile();
+                    file.setContents(FileUtils.readFileToString(new File(values[1])));
+                    file.setKeep(Boolean.valueOf(values[2]));
+                    job.getExtraFiles().put(values[0],file);
+                }
+                final Job workflowRun = jobApi.addOrder(job);
+                if (workflowRun == null){
+                    kill("consonance: failure reading back scheduled job");
+                }
+                outWithoutFormatting(serialize(workflowRun));
+            } catch (ApiException e) {
+                kill("consonance: could not schedule");
+            } catch (IOException e) {
+                kill("consonance: could not read file");
+            }
+        }
+    }
+
+    public static void main(String[] argv) throws IOException, TimeoutException, ApiException, OperationNotSupportedException {
+        List<String> args = new ArrayList<>(Arrays.asList(argv));
+        if (flag(args, "--debug")) {
+            DEBUG.set(true);
+        }
+        if (flag(args, "--verbose")) {
+            VERBOSE.set(true);
+            throw new OperationNotSupportedException("Not implemented yet");
+        }
+
+        if (isHelp(args, true)) {
+            out("");
+            out("Usage: consonance [<flag>]");
+            out("       consonance <command> [--help]");
+            out("");
+            out("Commands:");
+            out("  run           Schedule a job");
+            out("  status        Get the status of a job");
+            out("  update        Update this tool to a newer version");
+            // out("  dev           Advanced commands that are useful for developers or debugging");
+            out("");
+            out("Flags:");
+            out("  --help        Print help out");
+            // handled in seqware script:
+            out("  --version     Print Consonance's version");
+            out("  --metadata    Print metadata environment");
+            out("");
+        } else {
+            try {
+                String cmd = args.remove(0);
+                File configFile = new File(System.getProperty("user.home"), ".consonance/config");
+                final HierarchicalINIConfiguration hierarchicalINIConfiguration = Utilities.parseConfig(configFile.getAbsolutePath());
+                WebClient client = new WebClient(hierarchicalINIConfiguration);
+                client.setDebugging(DEBUG.get());
+
+                if (null != cmd) {
+                    switch (cmd) {
+                    case "-v":
+                    case "--version":
```

Comment:
Hmm maybe time to consider some Java CLI framework? Now's our chance to break away from JOpt (if we ever will). I recall that http://jcommander.org/ looked pretty cool, but never implemented it in SeqWare.